src/services/storage/handlers/offer.ts

```diff
@@ -75,7 +81,10 @@ const handlers: { [key: string]: Function } = {
 
     if (offerService.emit) {
       const freshOffer = await Offer.findByPk(offer.provider) as Offer
-      offerService.emit('updated', wrapEvent('BillingPlanSet', freshOffer.toJSON()))
+
+      if (freshOffer) {
+        offerService.emit('updated', wrapEvent('BillingPlanSet', freshOffer.toJSON()))
```

Comment:
Hmm should we emit `deleted` for when Billing Price is removed? 🤔

Reply:
Is it already sent by feather service by default?

Reply:
It would be if you would use `offerService.remove()` but not when you remove it using the Sequelize's model directly.

Reply:
Yes, you are right. But anyway we do not have BillingPlan service to emit `remove` event for billing plan

Reply:
Do we really need that?